### DISCUSSION TITLE
Resume the msi installation after .NET framework 4.7.2 installation with a reboot

### DISCUSSION THREAD

**Discussion by rekhasasidharan** [wixtoolset/issues#6566] [Questions] [answered]
I am building an application with the wixinstaller which needs .NET Framework 4.7.2 as one of the prerequisites. Added a custom action to install the .netframework through the exe ndp472_kb4054531_web.exe, through a powershell script. Since the target machine doesn't have the internet connection, the exe is downloaded and included in the project, which is installing through the script. Now the scripts executes and installs the .NET framework 4.7.2 in the target machine and reboot the machine. Now I need to find a way to resume the installation after the reboot. Can anyone suggest an idea to move forward.

The call from the C# code is as

dotNetInstallScriptFile = tempDirectory.PathJoin(@"installDotnet.ps1");
var exitCode = StartProcess("Powershell.exe", $"-file "{dotNetInstallScriptFile}"");

The installDotnet.ps1 is

Start-Process -FilePath 'ndp472_kb4054531_web.exe' /passive -Wait -Passthru

I have tried out /norestart /q options in the script and the system reboot was not happened with out the .NET framework installation. Thank you.

**Comment by robmen** [comment, marked as answer]
Burn (the engine behind Bundles) does this. You could use a Bundle (and Burn will do the work for you) or look at the code in Burn that handles it and replicate that in your code.

**Reply by rekhasasidharan** [reply]
Thanks Robmen for the response.  

Could you let me know that the bundling solution work in a environment where the internet is not connected. Another option which I tried out to resolve the above mentioned issue is by adding the the RunOnce registry key, which resumes the installation after the reboot of .NET Framework installation.  The key is removed after the reboot.

**Reply by BMurri** [reply]
If you run burn with the command-line argument `-layout {directory}` it will download and position all authored package files correctly to ensure that they will be available in non-internet-connected scenarios (those positions are relative to the bundle exe).
